src/channeldef.ts

```diff
@@ -494,80 +500,62 @@ export interface OrderFieldDef<F extends Field> extends FieldDefWithoutScale<F>
 
 export interface StringFieldDef<F extends Field> extends FieldDefWithoutScale<F, StandardType>, FormatMixins {}
 
-export type FieldDef<F extends Field> = SecondaryFieldDef<F> | TypedFieldDef<F>;
-export type ChannelDef<
-  FD extends FieldDef<any> = FieldDef<string>,
-  V extends ValueOrGradientOrText = ValueOrGradientOrText
-> = ChannelDefWithCondition<FD, V>;
+export type FieldDef<F extends Field, T extends Type = any> = SecondaryFieldDef<F> | TypedFieldDef<F, T>;
+export type ChannelDef<F extends Field = string> = Encoding<F>[keyof Encoding<F>];
 
-export function isConditionalDef<F extends Field, V extends ValueOrGradientOrText>(
-  channelDef: ChannelDef<FieldDef<F>, V>
-): channelDef is ChannelDefWithCondition<FieldDef<F>, V> {
+export function isConditionalDef<CD extends ChannelDef<any> | GuideEncodingConditonalValueDef>(
+  channelDef: CD
+): channelDef is CD & {condition: Conditional<any>} {
   return !!channelDef && !!channelDef['condition'];
 }
 
 /**
  * Return if a channelDef is a ConditionalValueDef with ConditionFieldDef
  */
-export function hasConditionalFieldDef<F extends Field, V extends ValueOrGradientOrText>(
-  channelDef: ChannelDef<FieldDef<F>, V>
-): channelDef is Partial<ValueDef<V>> & {condition: Conditional<TypedFieldDef<F>>} {
-  return !!channelDef && !!channelDef.condition && !isArray(channelDef.condition) && isFieldDef(channelDef.condition);
+export function hasConditionalFieldDef<F extends Field>(
+  channelDef: ChannelDef<F>
+): channelDef is {condition: Conditional<TypedFieldDef<F>>} {
+  const condition = channelDef && channelDef['condition'];
+  return !!condition && !isArray(condition) && isFieldDef(condition);
 }
 
-export function hasConditionalValueDef<F extends Field, V extends ValueOrGradient>(
-  channelDef: ChannelDef<FieldDef<F>, V>
-): channelDef is ValueDef<V> & {condition: Conditional<ValueDef<V>> | Conditional<ValueDef<V>>[]} {
-  return !!channelDef && !!channelDef.condition && (isArray(channelDef.condition) || isValueDef(channelDef.condition));
+export function hasConditionalValueDef<F extends Field>(
+  channelDef: ChannelDef<F>
+): channelDef is ValueDef<any> & {condition: Conditional<ValueDef<any>> | Conditional<ValueDef<any>>[]} {
+  const condition = channelDef && channelDef['condition'];
+  return !!condition && (isArray(condition) || isValueDef(condition));
 }
 
 export function isFieldDef<F extends Field>(
-  channelDef: ChannelDef<FieldDef<F>>
-): channelDef is
-  | TypedFieldDef<F>
-  | SecondaryFieldDef<F>
-  | PositionFieldDef<F>
-  | ScaleFieldDef<F>
-  | MarkPropFieldDef<F>
-  | OrderFieldDef<F>
-  | StringFieldDef<F> {
+  channelDef: ChannelDef<F>
+): channelDef is TypedFieldDef<F, any> | SecondaryFieldDef<F> {
```

Comment:
```suggestion
): channelDef is TypedFieldDef<F> | SecondaryFieldDef<F> {
```

It's already the default, no?